operators/src/processing/temporal_raster_aggregation/mean_aggregation_subquery.rs

```diff
@@ -0,0 +1,227 @@
+        let mut accu = accu;
+        accu.add_tile(tile)?;
+        Ok(accu)
+    })
+    .then(async move |x| match x {
+        Ok(r) => r,
+        Err(e) => Err(e.into()),
+    })
+}
+
+#[derive(Debug, Clone)]
+pub struct TemporalMeanTileAccu<T> {
+    time: TimeInterval,
+    tile_position: GridIdx2D,
+    global_geo_transform: GeoTransform,
+
+    value_grid: GridOrEmpty2D<f64>,
+    count_grid: Grid2D<u64>,
+    ignore_no_data: bool,
+    out_no_data_value: T,
+}
+
+impl<T> TemporalMeanTileAccu<T> {
+    pub fn add_tile(&mut self, in_tile: RasterTile2D<T>) -> Result<()>
+    where
+        T: Copy + AsPrimitive<f64> + Pixel,
+    {
+        self.time = self.time.union(&in_tile.time)?;
+
+        let in_tile_grid = match in_tile.grid_array {
+            GridOrEmpty::Grid(g) => g,
+            GridOrEmpty::Empty(_) => return Ok(()),
+        };
+
+        match &mut self.value_grid {
+            GridOrEmpty::Empty(_) => {
+                let mut accu_grid = self.value_grid.clone().into_materialized_grid();
+
+                for ((av, ac), iv) in accu_grid
+                    .data
+                    .iter_mut()
+                    .zip(self.count_grid.data.iter_mut())
+                    .zip(in_tile_grid.data.iter())
+                {
+                    if in_tile_grid.is_no_data(*iv) {
+                        *ac = 0;
+                    } else {
+                        let ivf: f64 = iv.as_();
+                        *av = ivf;
+                        *ac = 1;
+                    }
+                }
+
+                self.value_grid = accu_grid.into();
+            }
+
+            GridOrEmpty::Grid(accu_grid) => {
+                for ((av, ac), iv) in accu_grid
+                    .data
+                    .iter_mut()
+                    .zip(self.count_grid.data.iter_mut())
+                    .zip(in_tile_grid.data.iter())
+                {
+                    if in_tile_grid.is_no_data(*iv) {
+                        // The input pixel value is nodata
+                        if !self.ignore_no_data {
+                            // once nodata always nodata
+                            *ac = 0;
+                        }
+                    } else {
+                        let ivf: f64 = iv.as_();
+                        if self.ignore_no_data || *ac > 0 {
+                            // we either ignore nodata, then we add all non-nodata pixels or the count is > 0, so not nodata
+                            // *av += ivf;
+                            *ac += 1;
+                            let delta = ivf - *av;
+                            *av += delta / (*ac as f64);
+                        }
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+impl<T> FoldTileAccu for TemporalMeanTileAccu<T>
+where
+    T: Pixel,
+{
+    type RasterType = T;
+
+    fn tile_ref(&self) -> &RasterTile2D<Self::RasterType> {
+        unimplemented!();
```

Comment:
Dangerous, could crash our app.
Can't you throw some error or output something meanwhile and add a TODO?

Reply:
yes i was going to refactor the trait :(

Reply:
5708885